### PR TITLE
Update progressiveText.jelly with documentation tag

### PR DESCRIPTION
**Pull request by duemir**
The `st:documentation` and `st:attribute` tags are used by [Jenkins Development Support plugin](https://github.com/jenkinsci/idea-stapler-plugin/) for autocomplete and documentation lookup.

I did this quick change in the GitHub editor as much as a conversation starter as the suggestion. There more instances like this, so there is likely more work required.  Tags documented like this get a benefit of autocompletion and documentation and deprecated attribute validation from the IntelliJ plugin. The more tags are update with unsupported documentation format the less helpful IntelliJ plugin becomes.

See [JENKINS-XXXXX](https://issues.jenkins.io/browse/JENKINS-XXXXX).

### Proposed changelog entries

- human-readable text

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/resources/lib/hudson/progressiveText.jelly
Comment (timja) [inline core/src/main/resources/lib/hudson/progressiveText.jelly:43]: you've got mixed tabs and spaces in here

**Comment by duemir** [comment]
Sure, its flawed and needs to be fixed. But what about content? Does any tool rely on those JSP style comments? Or should I go forward and try and replace all of them with `st:documentation`/`st:attribute`?

**Comment by timja** [comment]
> Sure, its flawed and needs to be fixed. But what about content? Does any tool rely on those JSP style comments? Or should I go forward and try and replace all of them with `st:documentation`/`st:attribute`?

Not that I'm aware of, content looks good and yes good to use st:documentation etc everywhere

**Comment by duemir** [comment]
It turned out there were only a few. I addressed the comments and updated the other two tags.

**Review by timja — APPROVED** [review]
Thanks!

I think Alex went through at some point and fixed most of this up.

**Review by A1exKH — APPROVED** [review]
@duemir this PR LGTM.

**Review by krisstern — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by NotMyFault — APPROVED** [review]

**Review by A1exKH — APPROVED** [review]
@duemir LGTM.

**Review by Thompson1985 — APPROVED** [review]

**Pull request merged**
